x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/update_mappings.tsx

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { useValues } from 'kea';
+
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiTitle,
+  EuiText,
+  EuiPanel,
+  EuiBasicTableColumn,
+  EuiBasicTable,
+  EuiCallOut,
+  EuiLink,
+} from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+import { FieldMapping } from '../../../../../../../common/ml_inference_pipeline';
+
+import { docLinks } from '../../../../../shared/doc_links';
+
+import { MLInferenceLogic } from './ml_inference_logic';
+
+export const UpdateMappingsInstructions: React.FC = () => (
```

Comment:
Nit: Is it a good idea to split the components in this file into several smaller files, each containing just one component? Perhaps we could consider this as a technical debt task for the future. WDYT?

Reply:
It depends. This file in particular I wouldn't split because 1) it's self-contained (all subcomponents are referenced only from `UpdateMappings`, 2) it's mostly boilerplate code (sub-components have no or very simple decision logic), 3)  the overall file size is small.

In contrast, before the reorg `TextExpansionCallout` had 5 different subcomponents that took various properties, and it spanned 400+ lines of code.